kubermatic/dashboard
# Cluster Service Account UI (RBAC changes)

**What this PR does / why we need it**:
Added support for `service account` to cluster details RBAC

#### screenshots

<img width="217" alt="image" src="https://user-images.githubusercontent.com/56511988/205910970-5473ee0f-4a4c-4c35-aee3-95c256cd818c.png">

<img width="2472" alt="image" src="https://user-images.githubusercontent.com/56511988/205910244-5216459b-a668-4b93-aa5c-9b9be377068c.png">

<img width="2472" alt="image" src="https://user-images.githubusercontent.com/56511988/205910789-43bdf84f-d28c-41f6-83ca-30b70d8901cc.png">

<img width="2472" alt="image" src="https://user-images.githubusercontent.com/56511988/205910862-3916b806-b09b-41f4-831b-3573fc607d6b.png">

<img width="2466" alt="image" src="https://user-images.githubusercontent.com/56511988/205911103-d413db26-72e9-4dfd-b163-326e662a18f7.png">

<img width="2466" alt="image" src="https://user-images.githubusercontent.com/56511988/205911282-2bef59b3-de9e-4663-9a6a-2a60bcd44982.png">

<img width="666" alt="image" src="https://user-images.githubusercontent.com/56511988/205911439-f8315fdc-b172-44c7-be84-27e60f269ae5.png">

<img width="669" alt="image" src="https://user-images.githubusercontent.com/56511988/205911497-ffdb9517-5b0e-4629-bc5e-8d4bf39853bb.png">

<img width="669" alt="image" src="https://user-images.githubusercontent.com/56511988/205911531-a2637495-ff8d-4edc-bc34-9654e586b414.png">

<img width="669" alt="image" src="https://user-images.githubusercontent.com/56511988/205911553-ea110ac7-9bd4-45a9-bda4-45f8bb293886.png">


**Which issue(s) this PR fixes**:

Fixes #5292

**What type of PR is this?**
/kind feature


**Special notes for your reviewer**:
##### Testing service account
- If not pre-selected, select `service account` from the dropdown
 - Add a new `service account`
 - Observe `service account` being added on the table
 - Add binging to the `service account` from the modal
 - Observe the binding appear when clicked on the service account
 - `Deleting` service account's binding should remove it from the table
 - `Deleting` the service account should remove the whole row from the table

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Added support for service account the cluster
```

**Documentation**:

```documentation
TBD
```
